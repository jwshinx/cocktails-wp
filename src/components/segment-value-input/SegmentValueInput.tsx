import React from 'react'

import { SearchInput } from '../ui/SearchInput'
import { SegmentKeyType } from '../../models/cocktail'

const SegmentValueInput = ({
  onSegmentValueChange,
  segmentKey,
}: {
  segmentType: string
  segmentKey: SegmentKeyType
  onSegmentValueChange: (value: string) => void
}): JSX.Element => {
  const { name: segmentKeyName } = segmentKey
  const placeholderTxt = `${
    segmentKeyName === 'firstLetter' ? "(eg: first letter 'g')" : "(eg: 'gin')"
  }`

  return (
    <div className="row">
      <span>Enter value</span>
      <br />
      <SearchInput
        placeholderText={placeholderTxt}
        setSearchQuery={(query) => {
          onSegmentValueChange(query)
        }}
      />
    </div>
  )
}

export default SegmentValueInput
