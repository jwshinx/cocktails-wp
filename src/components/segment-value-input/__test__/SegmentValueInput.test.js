/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

const mockFn = jest.fn(() => true)

import SegmentValueInput from '../SegmentValueInput'

it('should render label "Enter value"', () => {
  const st = 'search'
  const sk = { label: 'xlabel', name: 'xname', value: 'f' }

  render(
    <SegmentValueInput
      segmentType={st}
      segmentKey={sk}
      onSegmentValueChange={mockFn}
    />
  )
  const elem = screen.getByText(/Enter value/)
  expect(elem).toBeInTheDocument()
})

it('should snapshot SegmentValueInput', () => {
  const st = 'search'
  const sk = { label: 'First Letter', name: 'firstLetter', value: 'f' }

  const tree = renderer
    .create(
      <SegmentValueInput
        segmentType={st}
        segmentKey={sk}
        onSegmentValueChange={mockFn}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

describe('SegmentValueInput', () => {
  it('should be able to type in input', () => {
    const st = 'search'
    const sk = { label: 'First Letter', name: 'firstLetter', value: 'f' }

    render(
      <SegmentValueInput
        segmentType={st}
        segmentKey={sk}
        onSegmentValueChange={mockFn}
      />
    )
    const elem = screen.getByPlaceholderText(/first letter/)
    fireEvent.change(elem, { target: { value: 'g' } })
    expect(elem.value).toBe('g')
  })
})
