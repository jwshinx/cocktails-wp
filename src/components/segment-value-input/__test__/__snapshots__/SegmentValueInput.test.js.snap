// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot SegmentValueInput 1`] = `
<div
  className="row"
>
  <span>
    Enter value
  </span>
  <br />
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="search"
      onChange={[Function]}
      placeholder="(eg: first letter 'g')"
      type="search"
    />
    <small
      className="form-text text-muted"
      id="emailHelp"
    />
  </div>
</div>
`;
